### ISSUE TITLE
Alternate file handling based on hostname doesn't work on Arch

### ISSUE THREAD

**Issue by emoryy** [TheLocehiliosan/yadm#15]
yadm uses `hostname -s` for determining the hostname, however this returns `localhost` on all of my Arch installs. This causes the alternate file handling to not work.

The `/etc/hostname` file on each of my systems contains the hostname on a single line.

I also put the hostname into the `/etc/hosts` file just like it is recommended in the [Arch wiki](https://wiki.archlinux.org/index.php/Network_configuration#Set_the_hostname)

I also tried `hostname -s` on some Ubuntu installs I have access to, and it interestingly returns the real hostname correctly.

Also simply calling `hostname` returns the hostname correctly both on Arch and Ubuntu.

**Comment by TheLocehiliosan** [comment]
I tried on an Arch host, and `hostname -s` works properly for me.

@emoryy, is it possible you have a misconfiguration? From what you wrote, I'm guessing you appended the hostname to an existing line in `/etc/hosts` for `127.0.0.1`. Have you tried to add an additional `127.0.0.1` line `/etc/hosts` for your hostname?

What result do you get when you run `getent hosts`?

**Comment by emoryy** [comment]
Yes, I merely did append the hostname to the end of the existing line, just like it is shown in the wiki.
Now I tried to add it to different lines too, like this:
```
#<ip-address>	<hostname.domain.org>	<hostname>
127.0.0.1	localhost.localdomain	my_hostname
::1		localhost.localdomain	my_hostname
127.0.0.1	localhost.localdomain	localhost
::1		localhost.localdomain	localhost
``` 

`hostname -s` still only returns localhost.
I also tried putting the two lines with `my_hostname` last, it didn't make a difference.
(I restarted my computer after each change fully.)

`getent hosts` returns this:

```
127.0.0.1       localhost.localdomain my_hostname
127.0.0.1       localhost.localdomain my_hostname
127.0.0.1       localhost.localdomain localhost
127.0.0.1       localhost.localdomain localhost
```

I don't know whether I have a misconfiguration or not... I installed and set up Arch on two desktop computers following the wiki. I also have Arch Linux Arm running on my Raspberry Pi Zero, and it does the same regarding `hostname -s`

**Comment by TheLocehiliosan** [comment]
Have you tried like this

    #<ip-address>   <hostname.domain.org>   <hostname>
    127.0.0.1   localhost.localdomain   localhost
    ::1     localhost.localdomain   localhost
    127.0.0.1   my_hostname my_hostname.localdomain

**Comment by TheLocehiliosan** [comment]
also... setting the hostname using `hostnamectl`.

**Comment by emoryy** [comment]
Yes, the solution is something like that. Thanks for the help.
I written out the following before your latest comments:

Ok, I looked around on the wiki a little bit more. On the same page that I linked in my first comment, [but in the "Local network hostname" resolution section] (https://wiki.archlinux.org/index.php/Network_configuration#Local_network_hostname_resolution) it shows this line:
```127.0.1.1	myhostname.localdomain	myhostname```

Note that `myhostname` is written in the second column too. Now, that wasn't the case within the "Set the hostname" section. Which section is also directly linked in the Beginners Guide.

Changed my `/etc/hosts` like that, now `hostname -s` return the real hostname.

So I guess maybe the wiki should be edited a little...
